src/panels/config/integrations/integration-panels/zha/zha-network-visualization-page.ts

```diff
@@ -0,0 +1,237 @@
+        if (devices[0]) {
+          navigate(
+            this,
+            "/config/devices/device/" + devices[0].device_reg_id,
+            false
+          );
+        }
+      }
+    });
+    this._firstUpdatedCalled = true;
+  }
+
+  protected render() {
+    return html`
+      <hass-subpage
+        .hass=${this.hass}
+        .header=${this.hass.localize(
+          "ui.panel.config.zha.visualization.header"
+        )}
+      >
+        <ha-config-section .isWide=${!this.narrow}>
+          <div id="visualization"></div>
+        </ha-config-section>
+      </hass-subpage>
+    `;
+  }
+
+  private async _fetchData() {
+    this._devices = await fetchDevices(this.hass!);
+    this._updateDevices(this._devices);
+  }
+
+  private _updateDevices(devices: ZHADevice[]) {
+    const nodes: Node[] = [];
+    const edges: Edge[] = [];
+
+    devices.forEach((device) => {
+      nodes.push({
+        id: device.ieee,
+        label: this._buildLabel(device),
+        shape: this._getShape(device),
+        mass: this._getMass(device),
+      });
+      if (device.neighbors && device.neighbors.length > 0) {
+        device.neighbors.forEach((neighbor) => {
+          const idx = edges.findIndex(function (e) {
+            return device.ieee === e.to && neighbor.ieee === e.from;
+          });
+          if (idx === -1) {
+            edges.push({
+              from: device.ieee,
+              to: neighbor.ieee,
+              label: neighbor.lqi + "",
+              color: this._getLQI(neighbor.lqi),
+            });
+          } else {
+            edges[idx].color = this._getLQI(
+              (parseInt(edges[idx].label!) + neighbor.lqi) / 2
+            );
+            edges[idx].label += "/" + neighbor.lqi;
+          }
+        });
+      }
+    });
+
+    this._network.setData({ nodes: nodes, edges: edges });
+  }
+
+  private _getLQI(lqi: number): EdgeOptions["color"] {
+    if (lqi > 192) {
+      return { color: "#17ab00", highlight: "#17ab00" };
+    }
+    if (lqi > 128) {
+      return { color: "#e6b402", highlight: "#e6b402" };
+    }
+    if (lqi > 80) {
+      return { color: "#fc4c4c", highlight: "#fc4c4c" };
+    }
+    return { color: "#bfbfbf", highlight: "#bfbfbf" };
+  }
+
+  private _getMass(device: ZHADevice) {
+    if (device.device_type === "Coordinator") {
+      return 2;
+    }
+    if (device.device_type === "Router") {
+      return 4;
+    }
+    return 5;
+  }
+
+  private _getShape(device: ZHADevice): string {
+    if (device.device_type === "Coordinator") {
+      return "box";
+    }
+    if (device.device_type === "Router") {
+      return "ellipse";
+    }
+    return "circle";
+  }
+
+  private _buildLabel(device: ZHADevice): string {
+    const regDevices = this._devices.filter((regDev) => {
+      return regDev.ieee === device.ieee;
+    });
+
+    let label =
+      regDevices.length > 0 && regDevices[0].user_given_name !== null
+        ? "<b>" + regDevices[0].user_given_name + "</b>\n"
```

Comment:
```suggestion
        ? `<b>${regDevices[0].user_given_name}</b>\n`
```

Do this for the others too.

Reply:
done